### PR TITLE
✅ Added tests for starknet_simulate_transactions and bump starknet.rs version to maintain compatibility with Deoxys

### PR DESCRIPTION
**Pull request by Tbelleng**
**Adding following tests for simulate_transactions call** 

1. fail_non_existing_block
2. fail_max_fee_too_big
3. fail_if_one_txn_cannot_be_executed

4. works_ok_on_no_validate
5. works_ok_on_validate_with_signature
6. works_ok_on_validate_without_signature_with_skip_validate

### CODE REVIEW

File: unit_tests/tests/test_simulate_transaction.rs [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+// # Returns
+// * `simulated_transactions` - The execution trace and consuemd resources of the required transactions
+//
+// # Errors
+// * `block_not_found` - If the block is not found or invalid
+// * `transaction_execution_error` - If one of the transactions failed to execute
+
+type RpcAccount<'a> = SingleOwnerAccount<&'a JsonRpcClient<HttpTransport>, LocalWallet>;
+
+pub fn build_single_owner_account<'a>(
```
Comment (antiyro) [inline unit_tests/tests/test_simulate_transaction.rs:43]: can you migrate all those to lib.rs

Comment (Tbelleng) [reply unit_tests/tests/test_simulate_transaction.rs]: Done sir ✅

File: unit_tests/tests/test_simulate_transaction.rs [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+    let pathfinder_simulations = pathfinder
+        .simulate_transactions(
+            BlockId::Tag(BlockTag::Latest),
+            &[invoke_transaction_pathfinder],
+            [],
+        )
+        .await
+        .unwrap();
+
+    assert_eq!(deoxys_simulations.len(), pathfinder_simulations.len());
```
Comment (antiyro) [inline unit_tests/tests/test_simulate_transaction.rs:201]: could you assert on the whole struct

**Comment by Tbelleng** [comment]
Last commit resolve #30

File: unit_tests/src/lib.rs [hunk 3]
```diff
@@ -1,10 +1,14 @@
 #![feature(assert_matches)]
 
 use constants::*;
+use starknet::signers::{LocalWallet, SigningKey};
```
Comment (antiyro) [inline unit_tests/src/lib.rs:4]: please set starknet_signers

File: unit_tests/Cargo.toml [hunk 4]
```diff
@@ -9,12 +9,13 @@ edition = "2021"
 anyhow = "1.0.79"
 rstest = "0.18.2"
 tokio = { version = "1", features = ["full"] }
-starknet = { git = "https://github.com/xJonathanLEI/starknet-rs.git", rev = "64ebc36", default-features = false }
-starknet-core = { git = "https://github.com/xJonathanLEI/starknet-rs.git", rev = "64ebc36", default-features = false }
-starknet-providers = { git = "https://github.com/xJonathanLEI/starknet-rs.git", rev = "64ebc36", default-features = false }
+url = "2.5.0"
+starknet = { git = "https://github.com/xJonathanLEI/starknet-rs.git", rev = "96c6803", default-features = false }
```
Comment (antiyro) [inline unit_tests/Cargo.toml:13]: remove this pls we wont to aim directly to a defined crate

**Pull request merged**